File: android-app/app 2/src/main/java/com/bandite/sonicwalkscape/data/models/AudioSettings.kt
package com.bandite.sonicwalkscape.data.models

data class AudioSettings(
    val volume: Float = 0.8f,
    val autoPlay: Boolean = true,
    val backgroundAudioEnabled: Boolean = true,
    val spatialAudioEnabled: Boolean = false
) {
    companion object {
        val DEFAULT = AudioSettings()
    }
}

File: mobile-app/ios/SonicWalkscape/SonicWalkscape/Models/AudioSettings.swift
import Foundation

struct AudioSettings: Codable {
    var volume: Float = 0.8
    var autoPlay: Bool = true
    var backgroundAudioEnabled: Bool = true
    var spatialAudioEnabled: Bool = false

    init(volume: Float = 0.8, autoPlay: Bool = true, backgroundAudioEnabled: Bool = true, spatialAudioEnabled: Bool = false) {
        self.volume = volume
        self.autoPlay = autoPlay
        self.backgroundAudioEnabled = backgroundAudioEnabled
        self.spatialAudioEnabled = spatialAudioEnabled
    }
}
